### ISSUE TITLE
[bug] CalDav Sync will not work with Outlook

### ISSUE THREAD

**Issue by acsinsurancegroup** [YetiForceCompany/YetiForceCRM#9859]
# 🐞 bug report

### 🔥 How to trigger the error

1. Go to 'Open my Outlook on my Laptop'
2. Click on 'Synchronize now'
3. Scroll down to 'N/A'
4. See error - Will upload Screenie

### 👎 Actual Behavior
Our CRM and Outlook will not Sync without errors

### 👍 Expected Behavior
For our CRM and Outlook to Sync without errors

### 🌍  Your Environment 

| Environment		|	Version / Name	|
| ----------------	|	--------------	|
| YetiForce			| 5.0
| Web server (name and version)|	Plesk Onyx 17.8.11 Update #44	|
| PHP  |		PHP 7.2.15 FPM 						|
| Browser (name and version)  | 	All & Latest on all Browsers I use	| 
| Operating System (name and version)  | 	CentOS 7	| 
| Database (name and version) | 	MariaDB 	| 

### ❓ Additional context
@setras @mariuszkrzaczkowski Hello again guys. As requested. here is the issue pertaining to our issue specifically. I will post a screenie of the logs as well and as requested by Marius.

So bottom line, I am connecting successfully, but it will not Sync no matter what. As you will see in the server logs, some of the highlighted stuff in pink, I will address below. I'm using Outlook CalDav Synchronizer to accomplish this: https://caldavsynchronizer.org So just trying to sync the CRM calendar with my desktop version of Microsoft Outlook with Microsoft Office.

The settings I am having issues with are: Some are actually set correctly, but the system won't acknowledge that they are set correctly.
<ul><li>opcache.interned_strings_buffer says it should be 100 but when I change it, it gives me errors and won't allow me to keep that setting.</li>
<li>disable_functions CRM says shell_exec, exec, system, passthru and I set it that way but logs still won't change</li>
<li>error_reporting this one is really frustrating because it has always been set as recommended yet logs say otherwise. I have it set to E_ALL & ~E_NOTICE despite what logs say.</li>
<li>zlib.output_compression same as error reporting. I have it set to off but logs says it's on in WWW so and Off in Cron?</li>
<li>date.timezone same issue. I have set it as noted and tried several formats of setting and it still always says it's wrong. I have tried GMT -5 / '' / - / None / left it blank and no matter what it shows an error.</li></ul>

@setras has tried to help me and I would be nowhere near as close as I am today without his help, so @setras again my friend, God bless and thank you so very much for all of your help.

@mariuszkrzaczkowski thanks in advance for your willingness to assist. Screenshots will be added in my own replies to this post. Thanks again

**Comment by acsinsurancegroup** [comment]
Here is the server log:

![yetiforce_settings](https://user-images.githubusercontent.com/42642523/54170994-dd237a00-4446-11e9-841f-7c0dbda2fda9.png)

**Comment by acsinsurancegroup** [comment]
Here's some of the Sync error information I'm getting with CalDav Synchronizer

![syncerror](https://user-images.githubusercontent.com/42642523/54171564-c847e600-4448-11e9-87b5-75bc41bcee0a.png)

**Comment by mariuszkrzaczkowski** [comment]
attach logs from CRM,
check this address, www. your_address_crm/**dav.php**
e.g. https://gitdeveloper.yetiforce.com/dav.php

![image](https://user-images.githubusercontent.com/10198654/54182199-17216a00-44a1-11e9-97b6-672344c43597.png)

**Comment by acsinsurancegroup** [comment]
@mariuszkrzaczkowski Thanks for reply. You stated: attach logs from CRM, but I thought that's what I posted above? 

Also, going to www.your_address_crm/dav.php does give me the login prompt. However, I cannot login to that link at all.

I can however, log in to this link without any issues: /mycrm/public_html/dav.php and it takes me into Sabre/Dav 3.2.2

The screenshot you posted, I'm not sure what I'm supposed to do with that info. As i stated above:

"date.timezone same issue. I have set it as noted and tried several formats of setting and it still always says it's wrong. I have tried GMT -5 / '' / - / None / left it blank and no matter what it shows an error."

I have tried every possible setting there :)

Thanks @mariuszkrzaczkowski

**Comment by mariuszkrzaczkowski** [comment]
Your cron works with PHP which has an incorrect configuration of the time zone and this can cause many eroros in operations on dates.

**Comment by setras** [comment]
you have to change the timezone setting in php.ini.
here a link how to change it in plesk 
https://support.plesk.com/hc/en-us/articles/115000494905-There-is-no-timezone-specified-in-domain-s-phpinfo

Link to the possible timezone settings
http://www.php.net/manual/en/timezones.php

**Comment by acsinsurancegroup** [comment]
@mariuszkrzaczkowski Ok Marius I finally got that part straight thanks to you and @setras and when I log into the CRM and the warnings show up, the Date and Timezone warning is no longer there so that issue is resolved. Yay, thanks guys. Now I will test syncing again and see if that resolves the issues I was having.

The only errors now showing up when the system warnings pop up are these:

Stability:
  error_reporting = 'www: E_ALL & ~E_NOTICE|cron: E_ALL & ~E_NOTICE' (Recommended: 'E_ALL & ~E_NOTICE')

I have this set correctly in the PHP.INI file but it still says error in warnings. Dunno how to clear that up.

  zlib.output_compression = 'www: On|cron: Off' (Recommended: 'Off')

I have this set correctly in the PHP.INI file also, but just like above error, it still shows up as an error in the warnings. Like above issue, I have no clue how to get the error message to go away.

Performance:
  opcache.interned_strings_buffer = 'www: 8|cron: 8' (Recommended: '100')

This error I mentioned in the previous thread. I cannot change it to 100 for some reason. Every time I do, it literally crashes the server and it won't work at all. I don't know if there's another setting I'm supposed to adjust with it as well, but all of the other recommended cache changes were completed previously without error. This one however, I just can't get to work. For now, I just have it commented out.

Also @setras Thanks for clarifying that for me. I have enabled the debugging and set the options you mentioned to True. I will post the results if I continue having issues now that the Date and Timezone issues has been corrected. I'm hoping, it will no longer be an issue now.  

Will keep you posted with my test results after I attempt another sync. Fingers crossed and thank you both very much.

**Comment by setras** [comment]
just saw the announcement for Version 5.1.  https://github.com/YetiForceCompany/YetiForceCRM/issues/9914#issue-420892689
In the change log there are two fixes that may help you.
- Fixed ignoring system alerts
- Improved DAV synchronization

**Comment by acsinsurancegroup** [comment]
@setras I cringe every time there is an update like this. I was hesitant, but I ran the update and verified that I am now running version 5.1 and I am.

It obviously didn't fix the ignoring of system alerts as I still show the following two errors:

Stability:
  zlib.output_compression = 'www: On|cron: Off' (Recommended: 'Off')

Despite this being set exactly as recommended, it still shows up as an error.

Performance:
  opcache.interned_strings_buffer = 'www: 8|cron: 8' (Recommended: '100')

I can't change this setting because as mentioned previously, it basically crashes my server.

Also when I click on show missing library in under System Warnings and Server Configuration it shows:

disable_functions | shell_exec, exec, system, passthru, popen | opcache_get_status
-- | -- | --

Even though I have it set to the recommended values.

And to the main issue at hand, I performed another sync using  Outlook CalDav Synchronizer and wouldn't you know, it still doesn't work. I've restarted all of my services, I've restarted the server itself and it still shows error. The connection tests are successful and last night I actually had two tests that showed up in Yellow color meaning they successfully ran, yet none since. I'm at a loss with this @setras it just doesn't make any sense.

I will post the server logs and see if @mariuszkrzaczkowski can provide additional guidance.

As always, thanks so much for your help :)

**Comment by acsinsurancegroup** [comment]
@mariuszkrzaczkowski Help, please :)

**Comment by acsinsurancegroup** [comment]
@setras and @mariuszkrzaczkowski OMG, I DID IT! IT WORKED GUYS :) SORRY FOR SHOUTING BUT I'M ECSTATIC. 

Ok so the issue was the PHP settings were set to serve over NGINX. I changed it to Apache and voila, it now works flawlessly. I setup a test entry on my Desktop version of Outlook. I then clicked on CalDav Synchronizer and hit Synchronize now and the test entry was in my CRM right after the next scheduled cron job ran. 

The meetings I add in the CRM however are NOT making it to my desktop version of Outlook which I would like to correct obviously, but for now I'm tickled Pink that it at least works from desktop to CRM. That's a huge start and a huge win.

By the way, I reverted the 5.1 update as well. To me it's buggy and the system is running way better just leaving it at 5.0 for now. 

Thanks fellas!

**Comment by setras** [comment]
@acsinsurancegroup congratulations!!

Check if the  https://caldavsynchronizer.org add-on is configured for two way snc.
https://caldavsynchronizer.org/help/documentation/

```
    Synchronization settings
        Outlook -> Server (Replicate): syncronize everything from Outlook to the server (one way)
        Outlook <- Server (Replicate): synchronize everything from the server to Outlook (one way)
        Outlook -> Server (Merge): synchronize everything from Outlook to the server but don’t change events created on the server
        Outlook <- Server (Merge): synchronize everything from the server to Outlook but don’t change events created in Outlook
        Outlook <-> Server (Two-Way): Two-Way synchronization between Outlook and the server with one of the following conflict resolution

```

**Comment by acsinsurancegroup** [comment]
@setras Thanks man, yes it was great to finally beat that thing lol.

I do have CalDav Sync setup that way, but it's just not sending from CRM to Desktop. So, for now I'm just setting up my appointments on Desktop and they are then coming into the system. If they're recurring events though, I'm having to setup the recurrence in the CRM as well because they don't come in as recurring. Other than that, at least it's working somewhat for now. Thanks again for all of your help. Not just on this but the Portal which also is working as well. I really appreciate it.

**Comment by acsinsurancegroup** [comment]
Thanks @mariuszkrzaczkowski for closing the issue, I forgot. My bad :)

**Comment by acsinsurancegroup** [comment]
@setras Hey buddy hope you're doing well. Ok so I have actually been meaning to mention this for a while just haven't had the time. It turned out it wasn't working 100% as I had originally thought. It really only worked maybe once or twice as far as syncing goes.

I did finally update to 5.1 again and everything is working great minus my calendar sync. I have tried CalDav Synchronizer and DAVx5 and the only one that occasionally works is DAVx5. You know how when you open up an email and select an invite.ics file and then it asks you which calendar you want to add that event to? Well, all my calendars show up on my phone and all have folders called **Events** and **Reminders** and some even have a folder called **Base Tasks, Birthdays & Holidays**.

Well my CRM calendar shows up as YFCalendar and has no folders whatsoever. There is no Events or Reminders or even Base Tasks. Is this why it doesn't really work for me? Am I supposed to create these specific folders in Savre Dav and if so are there some particular settings I'm supposed to use so this actually works right?

@mariuszkrzaczkowski Mariusz is there anything that you can suggest that would cure this issue for me. All I want is for my calendars to sync when I add an invite.ics file to my YFCalendar. Thanks guys :)

**Comment by acsinsurancegroup** [comment]
@setras @mariuszkrzaczkowski No matter what I do, every time I hit sync with DAVx5 I get the following error: Received invalid event from server and then it says this under the error Ignoring one or more invalid resources.

I have tried the Url setup with the following options and none work:
/public_html/dav.php
/public_html/dav.php/myemailaddress
/public_html/dav.php/principals/myemailaddress
/public_html/dav.php/principals/myemailaddress

Tried the URL's with and without https as well. The systems has no other issues whatsoever and I have green bars all throughout the system warnings area so no issues anywhere. Any thoughts guys?
